File: typescript/src/bracketIndex.ts
// Test cases covered: TC-55
// Source: EX-TS-BR-1

type User = { name: string; age: number };

const user: User = { name: "Ann", age: 21 };
// <caret> TC-55-A: Delete '"name"' below inside user[...], type 'na', invoke completion;
//   expect key completion for known property 'name'
const tc105a = user["name"];

const names = ["Ann", "Bob"];
const idx = 0;
// <caret> TC-55-B: Delete 'idx' below inside names[...], invoke completion;
//   expect numeric index variables from scope (e.g., idx)
const tc105b = names[idx];

void tc105a;
void tc105b;

export {};

File: python/bracket_index.py
"""
Bracket-index completion scenarios.

Covers: TC-101 (Python variant).
"""

users = [
    {"name": "Ann", "age": 21},
    {"name": "Bob", "age": 22},
]
idx = 0

# <caret> TC-101-A: Delete 'idx' below inside users[...], invoke completion;
#   expect numeric index variables from scope (e.g., idx)
tc106a = users[idx]

# <caret> TC-101-B: Delete '"name"' below inside users[idx][...], type 'na', invoke completion;
#   completion may suggest known dict keys in current context.
tc106b = users[idx]["name"]

_ = (tc106a, tc106b)
